daemon/main.go

```diff
@@ -721,7 +721,7 @@ func runDaemon() {
 	}
 
 	if err := d.EnableK8sWatcher(5 * time.Minute); err != nil {
-		log.WithError(err).Warn("Error while enabling k8s watcher")
+		log.WithError(err).Fatal("Error while enabling k8s watcher")
```

Comment:
Let's use a less cryptic error message here as this will be quite common:
`"Unable to establish connection to Kubernetes apiserver"`